packages/oauth/src/stores/file-store.spec.js

```diff
@@ -0,0 +1,262 @@
+require('mocha');
+const { assert, expect } = require('chai');
+
+const sinon = require('sinon');
+const fs = require('fs');
+
+const { InstallProvider } = require('../index');
+const { FileInstallationStore } = require('./index');
+
+const clientSecret = 'MY_SECRET';
+const clientId = 'MY_ID';
+const stateSecret = 'stateSecret';
+
+const storedInstallation = {
+  team: {
+    id: 'test-team-id',
+    name: 'team-name',
+  },
+  enterprise: {
+    id: 'test-enterprise-id',
+    name: 'ent-name',
+  },
+  bot: {
+    token: 'botToken',
+    scopes: ['chat:write'],
+    id: 'botId',
+    userId: 'botUserId',
+  },
+  user: {
+    token: 'userToken',
+    id: 'userId',
+  },
+  incomingWebhook: {
+    url: 'example.com',
+    channel: 'someChannel',
+    channelId: 'someChannelID',
+    configurationUrl: 'someConfigURL',
+  },
+  appId: 'fakeAppId',
+  tokenType: 'tokenType',
+  isEnterpriseInstall: false,
+}
+
+// TODO: valid tests with org-wide installations
+const storedOrgInstallation = {
+  team: null,
+  enterprise: {
+    id: 'test-enterprise-id',
+    name: 'ent-name',
+  },
+  bot: {
+    token: 'botToken',
+    scopes: ['chat:write'],
+    id: 'botId',
+    userId: 'botUserId',
+  },
+  user: {
+    token: 'userToken',
+    id: 'userId',
+  },
+  incomingWebhook: {
+    url: 'example.com',
+    channel: 'someChannel',
+    channelId: 'someChannelID',
+    configurationUrl: 'someConfigURL',
+  },
+  appId: undefined,
+  tokenType: 'tokenType',
+  isEnterpriseInstall: true,
+}
+
+describe('FileInstallationStore', async () => {
+
+  let fsMakeDir, fsWriteFile, fsReadFileSync;
+
+  beforeEach(() => {
+    fsMakeDir = sinon.stub(fs, 'mkdir').returns({});
+    fsWriteFile = sinon.stub(fs, 'writeFile').returns({});
+    fsReadFileSync = sinon.stub(fs, 'readFileSync').returns(Buffer.from(JSON.stringify(storedInstallation)));
+    fsUnlink = sinon.stub(fs, 'unlink').returns({});
+    fsReaddirSync = sinon.stub(fs, 'readdirSync').returns(['app-latest', 'user-userId-latest']);
+  });
+
+  afterEach(() => {
+    fsMakeDir.restore();
+    fsWriteFile.restore();
+    fsReadFileSync.restore();
+    fsUnlink.restore();
+    fsReaddirSync.restore();
+  });
+
+  it('should store the latest installation', async () => {
+    const installationStore = new FileInstallationStore({ baseDir: './tmp' });
```

Comment:
Changed the base directory for outputs - previously the tests used to generate directories in current dir.

Reply:
Not necessary as a change but another option is to use [`os.tmpDir()`](https://nodejs.org/api/os.html#ostmpdir)

Reply:
Oh, nice! will update this part.